### PR TITLE
added new problem 'meow'

### PR DESCRIPTION
**Pull request by NessaCat**

### CODE REVIEW

**Comment by chistopher** [comment]
builds and is uploaded to `fun-testing` now

File: cp21/fun-contest/meow/executables/generator.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from pathlib import Path
+
+SAMPLES = [
+    ['sample1','3\nmeow\nmeeeeeoooow\nmeeooooooooooooow\n'],
+    ['testcase1','1\nm'+'e'*20+'o'*70+'w'],
+    ['testcase2','1\nm'+'e'*1000+'o'*1000+'w\n'],
+    ['testcase3','1\nm'+'e'*100000+'o'*100000+'w\n'],
+    ['testcase4','1\nm'+'e'*1000000+'o'*1000000+'w\n'],
```
Comment (PFischbeck) [inline cp21/fun-contest/meow/executables/generator.py:8]: Could you add testcases (maybe random ones) with a much higher `t`? Right now, this allows for very slow solutions that calculate something for every single testcase. As a result, `solution-slow.tle.cpp` is not slow at all.

Comment (NessaCat) [reply cp21/fun-contest/meow/executables/generator.py:8]: That's interesting as the slow answer needed almost a minute for the third testcase alone on my machine but I will definitely change it then :)

Comment (PFischbeck) [reply cp21/fun-contest/meow/executables/generator.py:8]: Thanks! For reference, here's a run on my machine:

> $ make time-full-solution-slow.tle.cpp
Timing solution-slow.tle.cpp
sample1.in:                   0.005s
testcase1.in:                 0.004s
testcase2.in:                 0.004s
testcase3.in:                 0.005s
testcase4.in:                 0.017s
Maximum runtime: 0.017s

It is roughly the same on our server.

Comment (chistopher) [reply cp21/fun-contest/meow/executables/generator.py:8]: @NessaCat did you build release?

File: cp21/fun-contest/meow/problem.tex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The input starts with a line containing a single integer \textit{t} 
+\begin{math}(1\leq t\leq 10^4)\end{math}
+ the number of test cases that follow.
+
+Each test case consists of one line, each containing 1 string \textit{s}. The length of the string is between\begin{math}(4\leq length\leq 2*10^6+2)\end{math}.
+The string is the \textit{meow} of your cat, that is guaranteed to have each of the letters m, e, o, w at least one time.
+\paragraph*{Output}
+
+For each test case print the politeness of your cat and it's \textit{meow}.
```
Comment (chistopher) [inline cp21/fun-contest/meow/problem.tex:24]: was called happiness in text

File: cp21/fun-contest/meow/problem.tex [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+During social distancing, you and your cat spent way too much time together. It is very annoyed by your presence by now and it's \textit{meow}s are getting furious.
+To make it less difficult to live with your grumpy cat, you wanted to find a way to figure out when it is extremely unfriendly and annoyed, so you can take precautions.
+Many furious \textit{meow}s later, you started to notice a pattern.
+
+
+When you add the number of \textit{e}'s and \textit{o}'s in the \textit{meow} of your cat together, you can calculate its happiness by counting all possible ways this number \textit{n} can be expressed as the sum of consecutive integers larger than zero.
+
+\paragraph*{Input}
+
+The input starts with a line containing a single integer \textit{t}
```
Comment (chistopher) [inline cp21/fun-contest/meow/problem.tex:16]: if you want multiple testcases you must bound `t*|s|` over all tests. `10^10` is too much.

File: cp21/fun-contest/meow/problem.tex [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+When you add the number of \textit{e}'s and \textit{o}'s in the \textit{meow} of your cat together, you can calculate its happiness by counting all possible ways this number \textit{n} can be expressed as the sum of consecutive integers larger than zero.
+
+\paragraph*{Input}
+
+The input starts with a line containing a single integer \textit{t} 
+\begin{math}(1\leq t\leq 10^4)\end{math}
+ the number of test cases that follow.
+
+Each test case consists of one line, each containing 1 string \textit{s}. The length of the string is between\begin{math}(4\leq length\leq 2*10^6+2)\end{math}.
```
Comment (chistopher) [inline cp21/fun-contest/meow/problem.tex:20]: use `|s|` instead of length and `cdot` instead of `*`.

`4 \leq |s| \leq 2\cdot 10^6+2`

**Review by chistopher — COMMENTED** [review]
I just read the story and the problem seems gr8. Would be a bad contest if all problems were super hard.

**Comment by chistopher** [comment]
Thanks for your changes. I updated the problem on the judge again. Feel free to solve it (or any of the other problems) there.
However, the bounds still seem a bit off. `t` is not bounded. Also your stated bound for `|s|` (`4*10^6`) is factor 10 larger than the largest test case (`2*200'000 + 2).
As is, your generator could produce `50 * (2 * 200000 +2) = 20000100` output measures in `t * |s|`.
It would be sufficient if you stated that bound just as is: `t\cdot |s| \leq 20000100`

**Comment by chistopher** [comment]
@NessaCat Would you please address the issue of inconsistent bounds between generator and statement until Friday.

**Comment by NessaCat** [comment]
@chistopher Hey, I am not quite sure what and how to fix this. Is it a textual problem or a generator one?

**Comment by chistopher** [comment]
Purely textual.
1. you state a larger bounds for `|s|` than any testcase contains (`4*10^6` vs `4*10^5`)
2. you don't bound $t$ (or alternatively the sum of `|s|` over all testcases). Therefore it is within the given bounds to provide more input than could possibly be read in a second which would render this problem impossible.

My proposed solution to state the bound as `t\cdot |s| \leq 20000100` addresses both issues and is consistent with the maximum output of your generator under any given seed. However, in general you should always have a testcase that exactly matches the maximum bounds. First, this ensures that the runtime displayed to the user correctly reflects the speed they expect with your bounds and, second, it catches off-by-one errors of submissions that use statically sized data or constants.
Anyway, I would be fine with your generator as is, since this is your first problem and your generator is expected to produce 0.75 of its maximum output

**Pull request merged**